src/main/kotlin/org/cascadebot/cascadebot/data/objects/GuildData.kt

```diff
@@ -109,5 +138,27 @@ class GuildData(@field:Id val guildId: Long) {
 
     //endregion
 
+    fun write(writer: Consumer<GuildData>) {
+        this.lock.writeLock().withLock {
+            val copy: GuildData = CascadeBot.getGSON().fromJson(CascadeBot.getGSON().toJson(this), this.javaClass)
```

Comment:
Why... If we want to clone we should make a clone function of our own and not rely on GSON to do it properly!!

Reply:
https://github.com/CascadeBot/CascadeBot/pull/254#discussion_r642064489